tsl/src/fdw/data_node_scan_plan.c

```diff
@@ -462,32 +444,107 @@ add_data_node_scan_paths(PlannerInfo *root, RelOptInfo *baserel)
 		}
 	}
 
+	return ppi_list;
+}
+
+static void
+add_data_node_scan_paths(PlannerInfo *root, RelOptInfo *data_node_rel, RelOptInfo *hyper_rel,
```

Comment:
This the second part of the now-splitted function. It is called per data node and generates scan path.